lib/r10k/source/exec.rb

```diff
@@ -0,0 +1,45 @@
+require 'open3'
+require 'json'
+require 'yaml'
+
+class R10K::Source::Exec < R10K::Source::Hash
+  R10K::Source.register(:exec, self)
+
+  def initialize(name, basedir, options = {})
+    unless command = options[:command]
+      raise ConfigError, _('Environment source %{name} missing required parameter: command') % {name: name}
+    end
+
+    begin
+      stdout, stderr, status = Open3.capture3(command)
```

Comment:
Is there a use case for calling `open3` over using [`R10K::Util::Subprocess](https://github.com/puppetlabs/r10k/blob/master/lib/r10k/util/subprocess.rb)?

Reply:
Nope, just not familiar with R10K::Util::Subprocess. Modifying to use that...